ls1intum/Artemis
# `Adaptive learning`: Improve learning path sidebar in participation view

⚠️ Stacked on #7148  please don't merge accidentally
======



### Checklist
#### General


#### Client

### Motivation and Context



In order to make the UI in the participation view more intuitive, we want to replace the sidebar with a component that is tailored to the specific needs of a strictly linear representation of a learning path.

Resolves #7176 

### Description

- replaces the sidebar component with an improved implementation
- removed view encapsulation none from LearningPathGraphComponent

### Steps for Testing

Prerequisites:
- 1 Student
- 1 Course with a learning path enabled and at least one competency with many exercises

1. Log in to Artemis
2. Navigate to the Learning Paths tab in the course
3. check that the left sidebar is scrollable
4. check that the left sidebar is sticky (it does not disappear if you scroll down on a long exercise)
5. press the two up and down buttons, make sure the current lecture unit or exercise is highlighted and scrolled into view
6. press on the button on top of the sidebar
7. make sure the complete learning path graph opens and is displayed as expected

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage







#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [learning-path-graph.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5459/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/learning-paths/learning-path-graph/learning-path-graph.component.ts.html) | 87.8% | ✅  |
| [learning-path.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5459/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/learning-paths/learning-path-graph/learning-path.component.ts.html) | 97.14% | ✅  |
| [learning-path-container.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5459/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/learning-paths/participate/learning-path-container.component.ts.html) | 84.69% | ✅  |
| [learning-path-storage.service.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5459/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/learning-paths/participate/learning-path-storage.service.ts.html) | 96.29% | ✅  |
| [learning-path-progress-modal.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5459/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/learning-paths/progress-modal/learning-path-progress-modal.component.ts.html) | 100% | ✅ |
| [learning-path.model.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5459/latest/artifact/shared/Coverage-Report-Client-Tests/app/entities/competency/learning-path.model.ts.html) | 95.45% | ✅  |

 

### Screenshots


<img width="84" alt="Screenshot 2023-09-27 at 16 14 57" src="https://github.com/ls1intum/Artemis/assets/44003963/0863e9fa-a275-4ce3-bd97-3e52f5269265">
